sphinx/application.py

```diff
@@ -343,9 +343,14 @@ def build(self, force_all=False, filenames=None):
             status = (self.statuscode == 0 and
                       __('succeeded') or __('finished with problems'))
             if self._warncount:
-                logger.info(bold(__('build %s, %s warning.',
-                                    'build %s, %s warnings.', self._warncount) %
-                                 (status, self._warncount)))
+                if self.warningiserror:
+                    explanation = ' (with warnings treated as errors)'
+                else:
+                    explanation = ''
+                logger.info(bold(__('build %s, %s warning%s.',
```

Comment:
I _think_ you need to do the concatenation of strings before you log, otherwise the `explanation` string won't be translated. If I'm correct, you should probably do:

    msg = 'build %s, %s warning%s.'
    if self.warningiserror:
        msg += ' (with warnings treated as errors)'

    logger.info(bold(__(msg, msg, self.warn_count) % (status, self._warncount)))

But I'll wait for someone else more familiar with the translation framework to speak.

Reply:
our translation tool searches `__("...")` and extracts them as a translation message. So both `"build %s,..."` and `" (with warnings ..."` should be surrounded by `__(...)` function.

BTW, the concatenation is hard to translate because translators could not understand the context for each message (they don't read source code). So I prefer this:
```
                if self.warningiserror:
                    msg = __('build %s, %s warning (with warnings treated as errors).',
                             'build %s, %s warnings (with warnings treated as errors).',
                             self._warncount)
                else:
                    msg = __('build %s, %s warning.',
                             'build %s, %s warnings.',
                             self._warncount)

                logger.info(bold(msg % (status, self._warncount)))
```